test/specs/EventsSpec.js

```diff
@@ -131,167 +127,152 @@ describe("Event Tests", function() {
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'slide' event is triggered properly and can be binded to", function() {
+      it("'slide' event is triggered properly and can be binded to", function(done) {
         touch.initEvent("touchmove");
 
         testSlider.on('slide', function() {
-          flag = true;
+          done();
         });
         testSlider.data('slider')._mousemove(touch);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'slide' event sets the right value on the input", function() {
+      it("'slide' event sets the right value on the input", function(done) {
         touch.initEvent("touchmove");
 
         testSlider.on('slide', function() {
-          flag = true;
           expect(isNaN(testSlider.val())).not.toBeTruthy();
+          done();
         });
         testSlider.data('slider')._mousemove(touch);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'slide' event value and input value properties are synchronous", function() {
+      it("'slide' event value and input value properties are synchronous", function(done) {
         touch.initEvent("touchmove");
 
         testSlider.on('slide', function(e) {
-          flag = true;
           expect(e.value.toString()).toEqual(testSlider.val());
+          done();
         });
         testSlider.slider("setValue", 3, true, false);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'change' event value and input value properties are synchronous", function() {
+      it("'change' event value and input value properties are synchronous", function(done) {
         touch.initEvent("touchmove");
 
         testSlider.on('change', function(e) {
-          flag = true;
           expect(e.value.newValue.toString()).toEqual(testSlider.val());
+          done();
         });
         testSlider.slider("setValue", 3, false, true);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'slideStop' event is triggered properly and can be binded to", function() {
+      it("'slideStop' event is triggered properly and can be binded to", function(done) {
         touch.initEvent("touchstop");
 
         testSlider.on('slideStop', function() {
-          flag = true;
+          done();
         });
         testSlider.data('slider')._mouseup(mouse);
-
-        expect(flag).toBeTruthy();
       });
 
 
-      it("slider should not have duplicate events after calling 'refresh'", function() {
+      it("slider should not have duplicate events after calling 'refresh'", function(done) {
         touch.initEvent("touchstop");
-        flag = 0;
 
         testSlider.on('slideStop', function() {
-          flag += 1;
+          done();
         });
         testSlider.slider('refresh');
         testSlider.data('slider')._mouseup(mouse);
-
-        expect(flag).toEqual(1);
       });
 
       it("slider should not bind multiple touchstart events after calling 'refresh'", function(done) {
```

Comment:
This test makes sure that 'slideStart' is not called more than once, hence the `flag` variable.
Using a spy: (I thought spies were confusing, but they're very useful).
```js
it("slider should not bind multiple touchstart events after calling 'refresh'", function(done) {
    flag = 0;
    var obj = {
        addOne: function() {
        flag++;
        }
    };
    spyOn(obj, 'addOne').and.callThrough();

    touch.initEvent("touchstart", true, true);

    testSlider.on('slideStart', obj.addOne);

    testSlider.slider('refresh');
    $('#testSlider2').prev('div.slider').find('.slider-handle').get(0).dispatchEvent(touch);

    window.setTimeout(function() {
        expect(obj.addOne.calls.count()).toBe(1);
        done();
    }, 0);
});
```

Reply:
cool i will update